### PR TITLE
Rehydrate/Dehydrate Azure Blobs

### PR DESCRIPTION
**Pull request by mohsha-msft**

### CODE REVIEW

File: cmd/zt_scenario_helpers_for_test.go [hunk 1]
```diff
@@ -909,3 +909,28 @@ func getDefaultRemoveRawInput(src string) rawCopyCmdArgs {
 		includeDirectoryStubs:          true,
 	}
 }
+
+func getDefaultSetPropertiesRawInput(src string, accessTier common.BlockBlobTier) rawCopyCmdArgs {
+	fromTo := common.EFromTo.BlobNone()
+	srcURL, _ := url.Parse(src)
+
+	if strings.Contains(srcURL.Host, "file") {
```
Comment (mohsha-msft) [inline cmd/zt_scenario_helpers_for_test.go:917]: Make it just blob

File: common/fe-ste-models.go [hunk 2]
```diff
@@ -538,6 +539,8 @@ func (FromTo) BlobFile() FromTo    { return FromTo(fromToValue(ELocation.Blob(),
 func (FromTo) FileFile() FromTo    { return FromTo(fromToValue(ELocation.File(), ELocation.File())) }
 func (FromTo) S3Blob() FromTo      { return FromTo(fromToValue(ELocation.S3(), ELocation.Blob())) }
 func (FromTo) GCPBlob() FromTo     { return FromTo(fromToValue(ELocation.GCP(), ELocation.Blob())) }
+func (FromTo) BlobNone() FromTo    { return FromTo(fromToValue(ELocation.Blob(), ELocation.None())) }
+func (FromTo) FileNone() FromTo    { return FromTo(fromToValue(ELocation.File(), ELocation.None())) }
```
Comment (mohsha-msft) [inline common/fe-ste-models.go:543]: BlobFSNone

File: ste/mgr-JobPartMgr.go [hunk 3]
```diff
@@ -584,7 +584,7 @@ func (jpm *jobPartMgr) createPipelines(ctx context.Context) {
 	// Create pipeline for data transfer.
 	switch fromTo {
 	case common.EFromTo.BlobTrash(), common.EFromTo.BlobLocal(), common.EFromTo.LocalBlob(), common.EFromTo.BenchmarkBlob(),
-		common.EFromTo.BlobBlob(), common.EFromTo.FileBlob(), common.EFromTo.S3Blob(), common.EFromTo.GCPBlob():
+		common.EFromTo.BlobBlob(), common.EFromTo.FileBlob(), common.EFromTo.S3Blob(), common.EFromTo.GCPBlob(), common.EFromTo.BlobNone(), common.EFromTo.FileNone():
```
Comment (mohsha-msft) [inline ste/mgr-JobPartMgr.go:587]: Ditto

File: cmd/setPropertiesProcessor.go [hunk 4]
```diff
@@ -22,6 +22,25 @@ package cmd
 
 import "github.com/Azure/azure-storage-azcopy/v10/common"
 
+// setting SetPropertiesAPIOption for choosing which API to use
+// TODO name change needed
+func setSetPropertiesAPIOption(cca *CookedCopyCmdArgs) common.SetPropertiesAPIOption {
+	if cca.metadata != "" {
+		if cca.blockBlobTier != common.EBlockBlobTier.None() {
+			return common.ESetPropertiesAPIOption.SetTierAndMetaData()
```
Comment (mohsha-msft) [inline cmd/setPropertiesProcessor.go:30]: What if there are three or more properties. The best way to do it using bit mask. Look for ways to do it.

File: cmd/setProperties.go [hunk 5]
```diff
@@ -116,5 +116,6 @@ func init() {
 	setPropCmd.PersistentFlags().StringVar(&raw.listOfFilesToCopy, "list-of-files", "", "Defines the location of text file which has the list of only files to be copied.")
 	setPropCmd.PersistentFlags().StringVar(&raw.blockBlobTier, "block-blob-tier", "None", "Changes the access tier of the blobs to the given tier")
 	setPropCmd.PersistentFlags().BoolVar(&raw.recursive, "recursive", false, "Look into sub-directories recursively when uploading from local file system.")
-
+	setPropCmd.PersistentFlags().StringVar(&raw.metadata, "metadata", "", "Key-value pairs to be set as metadata. Pass empty string to delete all metadata.")
```
Comment (mohsha-msft) [inline cmd/setProperties.go:119]: We'll not expose this so this line is not required.

File: common/fe-ste-models.go [hunk 6]
```diff
@@ -119,6 +119,34 @@ func (DeleteSnapshotsOption) None() DeleteSnapshotsOption    { return DeleteSnap
 func (DeleteSnapshotsOption) Include() DeleteSnapshotsOption { return DeleteSnapshotsOption(1) }
 func (DeleteSnapshotsOption) Only() DeleteSnapshotsOption    { return DeleteSnapshotsOption(2) }
 
+type SetPropertiesAPIOption uint8
+
+var ESetPropertiesAPIOption = SetPropertiesAPIOption(0)
+
+func (SetPropertiesAPIOption) None() SetPropertiesAPIOption        { return SetPropertiesAPIOption(0) }
+func (SetPropertiesAPIOption) SetTier() SetPropertiesAPIOption     { return SetPropertiesAPIOption(1) }
+func (SetPropertiesAPIOption) SetMetaData() SetPropertiesAPIOption { return SetPropertiesAPIOption(2) }
```
Comment (tiverma-msft) [inline common/fe-ste-models.go:128]: Should this option be removed since we're not interested in Metadata as of now?

**Pull request closed without merging**